x/stakeibc/keeper/icacallbacks_detokenize.go

```diff
@@ -46,12 +46,22 @@ func DetokenizeCallback(k Keeper, ctx sdk.Context, packet channeltypes.Packet, a
 	k.Logger(ctx).Info(utils.LogICACallbackStatusWithHostZone(chainId, ICACallbackID_Detokenize,
 		icacallbackstypes.AckResponseStatus_SUCCESS, packet))
 
+	// Confirm host zone exists
+	hostZone, found := k.GetHostZone(ctx, chainId)
+	if !found {
+		return errorsmod.Wrapf(types.ErrHostZoneNotFound, "Host zone not found: %s", chainId)
+	}
+
 	// If the ICA succeeded, remove the token deposit
 	deposit := detokenizeCallback.Deposit
 	k.RecordsKeeper.RemoveLSMTokenDeposit(ctx, deposit.ChainId, deposit.Denom)
 
 	// Update delegation on the host zone and validator
-	// TODO [LSM] - Use helper function
+	err := k.AddDelegationToValidator(ctx, &hostZone, deposit.ValidatorAddress, deposit.Amount, ICACallbackID_Detokenize)
```

Comment:
This is the only place in the codebase I have seen this pattern (antipattern?) of passing the object by reference and updating it as a side effect in the function instead of just returning a changed version of the object.  Is it just because hostZone is big and complex or is there a reason to do it like this?  Why not also setHostZone inside the addDelegationToValidator then?

Reply:
Great question! I personally dislike this pattern, but decided on it for the following reasons:
a) most functions loop and modify multiple validators in a row so it's slightly more efficient to just do one HostZone set at the end (although this probably doesn't matter much)
b) more importantly, since this is often called from epochs (which runs in the EndBlocker), I was hesitant to Set prematurely as the state wouldn't get reverted (however, there is a function that we'll integrate soon that will let us do the reversion)
c) it was set up this way originally so, while it's not perfect right now, it's at least not worse haha

All that said, I think the ideal approach is to do (a) the refactor I mentioned before where we have a separate prefix store for validators and (b) add in the error catching in the EndBlocker to revert state - and then come back and clean this up once those are in place